metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/data_resolver.py

```diff
@@ -0,0 +1,660 @@
+        ]
+
+        return users
+
+    def get_scan_result(self, scan_id: str) -> dict:
+        logger.info("Fetching scan result")
+        logger.info(f"{Constant.SCAN_ID}={scan_id}")
+        scan_result_get_endpoint = AdminAPIResolver.API_ENDPOINTS[
+            Constant.SCAN_RESULT_GET
+        ]
+        scan_result_get_endpoint = scan_result_get_endpoint.format(
+            POWERBI_ADMIN_BASE_URL=DataResolverBase.ADMIN_BASE_URL, SCAN_ID=scan_id
+        )
+
+        logger.debug(f"Hitting URL={scan_result_get_endpoint}")
+        res = self._request_session.get(
+            scan_result_get_endpoint,
+            headers={Constant.Authorization: self.get_access_token()},
+        )
+        if res.status_code != 200:
+            message = f"API({scan_result_get_endpoint}) return error code {res.status_code} for scan id({scan_id})"
+            logger.warning(message)
+            raise ConnectionError(message)
+
+        return res.json()["workspaces"][0]
+
+    def get_groups_endpoint(self) -> str:
+        return f"{AdminAPIResolver.ADMIN_BASE_URL}/groups"
+
+    def get_dashboards_endpoint(self, workspace: Workspace) -> str:
+        dashboard_list_endpoint: str = self.API_ENDPOINTS[Constant.DASHBOARD_LIST]
+        # Replace place holders
+        return dashboard_list_endpoint.format(
+            POWERBI_ADMIN_BASE_URL=DataResolverBase.ADMIN_BASE_URL,
+            WORKSPACE_ID=workspace.id,
+        )
+
+    def get_reports_endpoint(self, workspace: Workspace) -> str:
+        reports_endpoint: str = self.API_ENDPOINTS[Constant.REPORT_LIST]
+        return reports_endpoint.format(
+            POWERBI_ADMIN_BASE_URL=DataResolverBase.ADMIN_BASE_URL,
+            WORKSPACE_ID=workspace.id,
+        )
+
+    def get_tiles_endpoint(self, workspace: Workspace, dashboard_id: str) -> str:
+        tiles_endpoint: str = self.API_ENDPOINTS[Constant.TILE_LIST]
+        # Replace place holders
+        return tiles_endpoint.format(
+            POWERBI_ADMIN_BASE_URL=DataResolverBase.ADMIN_BASE_URL,
+            DASHBOARD_ID=dashboard_id,
+        )
+
+    def get_dataset(
+        self, workspace_id: str, dataset_id: str
+    ) -> Optional[PowerBIDataset]:
+        datasets_endpoint = self.API_ENDPOINTS[Constant.DATASET_LIST].format(
+            POWERBI_ADMIN_BASE_URL=DataResolverBase.ADMIN_BASE_URL,
+            WORKSPACE_ID=workspace_id,
+        )
+        # Hit PowerBi
+        logger.info(f"Request to datasets URL={datasets_endpoint}")
+        params: dict = {"$filter": f"id eq '{dataset_id}'"}
+        logger.debug("params = %s", params)
+        response = self._request_session.get(
+            datasets_endpoint,
+            headers={Constant.Authorization: self.get_access_token()},
+            params=params,  # urllib.parse.urlencode(params, doseq=True).replace('+', '%20'),
+        )
+        response.raise_for_status()
+        response_dict = response.json()
+        if len(response_dict["value"]) == 0:
```

Comment:
What if response_dict["value"] is None? Error.

Reply:
Added guard condition